controllers/apim/ratelimitpolicy_limits.go

```diff
@@ -115,23 +118,35 @@ func (r *RateLimitPolicyReconciler) gatewayLimits(ctx context.Context, rlpRefs [
 		if rlp.IsForHTTPRoute() {
 			routeRLPList = append(routeRLPList, rlp)
 		} else if rlp.IsForGateway() {
-			gwRLPList = append(gwRLPList, rlp)
+			if gwRLP == nil {
+				gwRLP = rlp
+			} else {
+				return nil, fmt.Errorf("gatewayLimits: multiple gateway RLP found and only one expected. rlp keys: %v", rlpRefs)
+			}
 		}
 	}
 
 	limits := rlptools.LimitsByDomain{}
 
-	// iterate over HTTPRoute RLP's.
-	// Gateway level RLP's alone do not make sense.
-	// The request would be rejected by the router with 404
+	if gwRLP != nil {
+		if len(gw.Hostnames()) == 0 {
+			// wildcard domain
+			limits["*"] = append(limits["*"], gwRLP.FlattenLimits()...)
```

Comment:
I reckon this is because the Route RLP was copied to the GW RLP, but the former doesn't have in the Spec "Hostnames", so we apply all the limits to the arbitrary wildcard domain.. would that be correct?

Reply:
I'd be tempted to extract this logic in to a fn, which name would be descriptive of what it does

Reply:
we need to add gateway limits to the index which is indexed by the domain. If the gateway does not specify any domain, global wilcard domain is assumed. and the code creates a new entry with the gateway limits indexed by the wilcard domain `*`

Reply:
I see, makes sense, another way would be to have a validating webhook returning the error that the hostname is missing

Reply:
well, if the hostname is missing, there is nothing wrong with it. It is just equivalent to `*`